Objective-C/Tests/URLEndpointListenerTest.m

```diff
@@ -0,0 +1,162 @@
+//
+//  URLEndpointListenerTest.m
+//  CouchbaseLite
+//
+//  Copyright (c) 2020 Couchbase, Inc All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+#import "ReplicatorTest.h"
+#import "CBLURLEndpointListener.h"
+#import "CBLURLEndpointListenerConfiguration.h"
+#import "CollectionUtils.h"
+
+#define kPort 4984
+#define kURL [NSURL URLWithString: $sprintf(@"ws://localhost:%d/otherdb", kPort)]
+
+API_AVAILABLE(macos(10.12), ios(10.0))
+@interface URLEndpointListenerTest : ReplicatorTest
+typedef CBLURLEndpointListenerConfiguration Config;
+typedef CBLURLEndpointListener Listener;
+
+@end
+
+@implementation URLEndpointListenerTest
+
+- (Listener*) listenAt: (uint16)port {
+    Config* config = [[Config alloc] initWithDatabase: self.otherDB port: port];
+    config.disableTLS = YES;
+    Listener* list = [[Listener alloc] initWithConfig: config];
+    
+    // start listener
+    NSError* err = nil;
+    Assert([list startWithError: &err]);
+    AssertNil(err);
+    
+    return list;
+}
+
+- (void) testPort {
+    // initialize a listener
+    Config* config = [[Config alloc] initWithDatabase: self.otherDB port: kPort];
+    config.disableTLS = YES;
+    Listener* list = [[Listener alloc] initWithConfig: config];
+    AssertEqual(list.port, 0);
+    
+    // start listener
+    NSError* err = nil;
+    Assert([list startWithError: &err]);
+    AssertNil(err);
+    AssertEqual(list.port, kPort);
+    
+    // stops
+    [list stop];
+    AssertEqual(list.port, 0);
+}
+
+- (void) testEmptyPort {
+    // initialize a listener
+    Config* config = [[Config alloc] initWithDatabase: self.otherDB port: 0];
+    config.disableTLS = YES;
+    Listener* list = [[Listener alloc] initWithConfig: config];
+    AssertEqual(list.port, 0);
+    
+    // start listener
+    NSError* err = nil;
+    Assert([list startWithError: &err]);
+    AssertNil(err);
+    Assert(list.port != 0);
+    
+    // stops
+    [list stop];
+    AssertEqual(list.port, 0);
+}
+
+- (void) testBusyPort {
+    Listener* list1 = [self listenAt: kPort];
+    
+    // initialize a listener at same port
+    Config* config = [[Config alloc] initWithDatabase: self.otherDB port: kPort];
+    config.disableTLS = YES;
+    Listener* list2 = [[Listener alloc] initWithConfig: config];
+    
+    // already in use when starting the second listener
+    [self ignoreException:^{
```

Comment:
Why do we need to ignore the exception here?

Reply:
may be, the exception thrown from lite core might not be caught in between? will check

Reply:
already existing port will throw exception from liter core. so the problem lies in hitting the exception break point

Reply:
will keep this as same. and merging, will fix this if any issue exists.